File: ios/ObjectiveC/FibonacciSeries.m
//
//  FibonacciSeries.m
//  MyFibonacci
//
//  Created by Krunal Kevadiya on 03/09/23.
//

#import "FibonacciSeries.h"

@implementation FibonacciSeries

- (instancetype)init {
  self = [super init];
  if (self) {
    _num = @0;
    _num1 = @0;
    _num2 = @1;
  }
  return self;
}

- (NSString*) getInitFibonacciNum
{
  return [NSString stringWithFormat:@"%@ %@", self.num1, self.num2];
}

- (NSString*) getFibonacciNum
{
  return [NSString stringWithFormat:@" %@", self.num];
}

- (void) generateNextFibonacciNum
{
  self.num = [NSNumber numberWithLong:([self.num1 longValue] + [self.num2 longValue])];
  self.num1 = self.num2;
  self.num2 = self.num;
}

- (void) resetFibonacciNum
{
  _num = @0;
  _num1 = @0;
  _num2 = @1;
}
@end

File: ios/Swift/FibonacciSeries.swift
//
//  FibonacciSeries.swift
//  MyFibonacci
//
//  Created by Krunal Kevadiya on 03/09/23.
//

import Foundation

class FibonacciSeries {
  private var num: UInt64?
  private var num1: UInt64?
  private var num2: UInt64?
  
  init() {
    num = 0
    num1 = 0
    num2 = 1
  }
  
  public func getInitFibonacciNum() -> String {
    return "\(num1 ?? 0) \(num2 ?? 0)"
  }
  
  public func getFibonacciNum() -> String {
    return " \(num ?? 0)"
  }
  
  public func generateNextFibonacciNum() -> Void {
    num = (num1 ?? 0) + (num2 ?? 0)
    num1 = num2
    num2 = num
  }
  
  public func resetFibonacciNum() -> Void {
    num = 0
    num1 = 0
    num2 = 1
  }
}
